sdk/src/main/java/me/uport/sdk/Transactions.kt

```diff
@@ -78,45 +83,64 @@ class Transactions(
                 gasLimit = gasLimit)
     }
 
-
-    /**
-     * sends a given transaction to the Eth network.
-     * Depending on the [signerType], the transaction may be wrapped into some other transaction
-     *
-     */
     suspend fun sendTransaction(signer: Signer, request: Transaction, signerType: SignerType = Proxy): String {
-        val unsigned = buildTransaction(request, signerType)
+        val txLabel = request.encodeRLP().toHexString()
 
-        val relaySigner = TxRelaySigner(signer, network)
+        var (state, oldBundle) = if(progress.contains(txLabel)) {
+            (ProgressPersistence.PendingTransactionState.NONE to ProgressPersistence.PersistentBundle())
+        } else {
+            progress.restore(txLabel)
+        }
 
-        val txHash = when (signerType) {
-            MetaIdentityManager -> {
+        if (state === ProgressPersistence.PendingTransactionState.NONE) {
+            val unsigned = buildTransaction(request, signerType)
+            oldBundle = oldBundle.copy(unsigned = unsigned, ordinal = ProgressPersistence.PendingTransactionState.TRANSACTION_BUILT.ordinal)
+            state = ProgressPersistence.PendingTransactionState.TRANSACTION_BUILT
```

Comment:
To avoid duplication and potential mistakes, it's better to only set the state once and the use `state.ordinal` to construct the new bundle